.github/workflows/ci.yaml

```diff
@@ -146,14 +129,13 @@ jobs:
           key: $clippy-cache-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Generate lockfile
-      # XXX: `0.6.1` of `toml_datetime` is the last version with a MSRV that matches
-      # our current one. 
+      # XXX: Downgrade various dependencies to make them or their sub-dependencies MSRV compatible.
         run: |
           cargo generate-lockfile
-          cargo update -p toml_datetime --precise "0.6.1"
+          cargo update -p proc-macro-crate --precise "1.3.0"
+          cargo update -p regex --precise "1.9.6"
+          cargo update -p zbus --precise "3.14.1"
+          cargo update -p tokio --precise "1.29.1"
 
-      - uses: actions-rs/cargo@v1
-        name: Clippy MSRV
-        with:
-          command: clippy
-          args: --features=rt-tokio-crypto-rust --all-targets --all -- -D clippy::dbg_macro -D warnings -F unused_must_use
+      - name: Clippy MSRV
+        run: cargo clippy --features=rt-tokio-crypto-rust --all-targets --all -- -D clippy::dbg_macro -D warnings -F unused_must_use

```

Comment:
How about using [`cargo msrv verify`](https://github.com/foresterre/cargo-msrv)?

Reply:
This could work, but the outstanding problem here is that we need to wrangle the lockfile into a place where all dependencies compiled use versions compatible with the MSRV. This is what I was struggling to accomplish last night because of `zbus`'s deep tree.